lldb/include/lldb/Core/PluginManager.h

```diff
@@ -319,6 +319,23 @@ class PluginManager {
 
   static llvm::StringRef GetProcessPluginDescriptionAtIndex(uint32_t idx);
 
+  // GPU Process (separate category from CPU Process plugins)
+  // GPU process plugins are only discovered via
+  // ProcessElfGpuCore::LoadGpuCore() and do not participate in normal
+  // Process::FindPlugin() discovery
+  static bool RegisterGpuProcessPlugin(
```

Comment:
If we are going to use an the PluginManager for these, this should be named to indicate that. We might not want to limit this to GPUs only as any core file can contain another architecture. Maybe something like:
```
static bool RegisterCompanionCoreProcessPlugin(
```